target/linux/ath79/dts/qca9558_tl-archer-c7.dtsi

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca9557.dtsi"
+
+/ {
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x8000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	aliases {
+		led-status = &system;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		system: system {
+			label = "tp-link:blue:system";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		usb1 {
+			label = "tp-link:green:usb1";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+			trigger-sources = <&hub_port0>;
+			linux,default-trigger = "usbport";
+		};
+
+		usb2 {
+			label = "tp-link:green:usb2";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+			trigger-sources = <&hub_port1>;
+			linux,default-trigger = "usbport";
+		};
+
+		wlan2g {
+			label = "tp-link:blue:wlan2g";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+			linux,default-trigger = "phy1tpt";
+		};
+
+		qss {
+			label = "tp-link:blue:qss";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		wlan5g {
+			label = "tp-link:blue:wlan5g";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		rfkill: wifi {
+			linux,code = <KEY_RFKILL>;
+			linux,input-type = <EV_SW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	gpio-export {
```

Comment:
Please use fixed regulators instead of the gpio exports. Have a look at https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/ar9331_tl-wr703n_tl-mr10u.dtsi and https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/ar9331_tl-mr3020-v1.dts for examples.

Reply:
@mkresin Do you know if someting similar is also possible for the ath9k related gpio_ext_lna0 in https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/ar9344_tl-wdr4300.dtsi?

Reply:
@juppin No idea, sorry. Grep for devm_regulator_get or regulator_enable in the ath9k code. It should look similar to these upstream commit: https://github.com/openwrt/openwrt/blob/master/target/linux/generic/backport-4.14/030-v4.17-0001-usb-dwc2-add-support-for-host-mode-external-vbus-sup.patch

Reply:
@mkresin Does this driver have support for this? I cannot find a related property in https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/usb/usb-ehci.txt or anything regulator related in drivers/usb/host/*ehci*.c
If I understand it correctly, is this what I would be looking for? https://lkml.org/lkml/2018/3/1/200

Reply:
@SibrenVasse I failed to see that qca9558 doesn't have a chipidea usb controller/uses the chipidea usb driver.

Indeed, the patch you linked would add the devicetree property for the common ehci driver. But this patch/approach was [rejected by upstream](https://lkml.org/lkml/2018/3/6/424).

Long story short, disregard my comment! At the moment there is no better way to do it.

Reply:
Ah, so my suspicion was correct. Thanks for your reply!